metals/src/main/scala/scala/meta/internal/metals/Messages.scala

```diff
@@ -370,82 +369,52 @@ object Messages {
     }
   }
 
-  // Don't confuse this with the "multiple build tools that can be servers"
-  // message up above.  That one focuses not on multiple .bsp/<tool>.json
-  // entries, but rather having multiple build tools in a workspace that could
-  // potentially be a build server, whereas this one focuses on existing .bsp
-  // files that already exist.
-  object SelectBspServer {
+  object BspSwitch {
     case class Request(
         params: ShowMessageRequestParams,
-        details: Map[String, BspConnectionDetails]
+        mapping: Map[String, String]
     )
-    def message: String =
+
+    val message: String =
       "Multiple build servers detected, which one do you want to use?"
-    def isSelectBspServer(params: ShowMessageRequestParams): Boolean =
-      params.getMessage == message
-    def request(
-        candidates: List[BspConnectionDetails],
+
+    def chooseServerRequest(
+        possibleBuildServers: List[String],
         currentBsp: Option[String]
     ): Request = {
-      val params = new ShowMessageRequestParams()
-      params.setMessage(message)
-      params.setType(MessageType.Warning)
-      val details = mutable.Map.empty[String, BspConnectionDetails]
-      // The logic for choosing item names is a bit tricky because we want
-      // the following characteristics:
-      // - all options must be unique, we get a title string back from the
-      //   editor for which server the user chose.
-      // - happy path: title is build server name without noisy version number.
-      // - name conflicts: disambiguate conflicting names by version number
-      // - name+version conflicts: append random characters to the title.
-      val items = candidates.map { candidate =>
-        val currentlyUsing =
-          if (
-            currentBsp.exists(
-              _.toLowerCase == candidate.getName().toLowerCase()
-            )
-          )
-            " (currently using)"
-          else ""
-        val nameConflicts = candidates.count(_.getName == candidate.getName)
-        val title: String = if (nameConflicts < 2) {
-          candidate.getName + currentlyUsing
-        } else {
-          val versionConflicts = candidates.count { c =>
-            c.getName == candidate.getName &&
-            c.getVersion == candidate.getVersion
-          }
-          if (versionConflicts < 2) {
-            s"${candidate.getName} v${candidate.getVersion}"
+      val mapping = mutable.Map.empty[String, String]
+      val messageActionItems =
+        possibleBuildServers.map { bs =>
```

Comment:
```suggestion
        possibleBuildServers.map { buildServer =>
```
Saying `bs` in my  head sounds a bit weird :sweat_smile:

Reply:
haha yea of course, switched.